File: manager/src/main/java/ru/vladp/manager/frontend/emailgreeting/services/EmailCreatorService.java
package ru.vladp.manager.frontend.emailgreeting.services;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import ru.vladp.manager.frontend.emailgreeting.interfaces.EmailCreator;
import ru.vladp.manager.frontend.emailgreeting.model.dto.EmailDTO;
import ru.vladp.manager.frontend.security.model.User;

import java.util.ArrayList;
import java.util.List;

@Service
@RequiredArgsConstructor
public class EmailCreatorService {

    private final EmailCreator emailCreator;

    public List<EmailDTO> getEmailMessages(List<User> users) {
        List<EmailDTO> emailMessages = new ArrayList<>();

        for (User user : users) {
            EmailDTO emailDTO = new EmailDTO();
            emailDTO.setRecipientAddress(emailCreator.createEmailAddress(user));
            emailDTO.setTitle(emailCreator.createEmailTitle(user));
            emailDTO.setText(emailCreator.createEmailText(user));
            emailMessages.add(emailDTO);
        }
        return emailMessages;
    }
}

File: manager-scheduler/src/main/kotlin/ru/vladp/manager/scheduler/service/EmailCreatorService.kt
package ru.vladp.manager.scheduler.service

import ru.vladp.manager.scheduler.interfaces.EmailCreator
import ru.vladp.manager.scheduler.model.User
import ru.vladp.manager.scheduler.model.dto.EmailDTO
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class EmailCreatorService(@Autowired private val emailCreator: EmailCreator) {
    fun getEmailMessages(users: MutableList<User>): MutableList<EmailDTO> {
        val emailMessages: MutableList<EmailDTO> = mutableListOf()
        for (user in users) {
            if (user.tasks.size > 0) {
                val emailDTO = EmailDTO(
                    emailCreator.createEmailAddress(user),
                    emailCreator.createEmailTitle(user),
                    emailCreator.createEmailText(user)
                )
                emailMessages.add(emailDTO)
            }
        }
        return emailMessages
    }
}